versioning/lockfiles/configurations.rst

```diff
@@ -0,0 +1,191 @@
+.. _versioning_lockfiles_configurations:
+
+Multiple configurations
+=======================
+
+.. warning::
+
+    This is an **experimental** feature subject to breaking changes in future releases.
+
+In the previous section we managed just 1 configuration, for the default profile. In many applications,
+packages needs to be built with several different configurations, typically managed by different profile
+files.
+
+.. note::
+
+    This section continues with the previous example with the :ref:`versioning_lockfiles_introduction`.
+    The code used in this section, including a *build.py* script to reproduce it, is in the
+    examples repository: https://github.com/conan-io/examples. You can go step by step
+    reproducing this example while reading the below documentation.
+
+    .. code:: bash
+
+        $ git clone https://github.com/conan-io/examples.git
+        $ cd features/lockfiles/intro
+        # $ python build.py only to run the full example, but better go step by step
+
+
+Lets start in the *features/lockfiles/intro* of the examples repository, remove the previous packages,
+and create both release and debug ``pkga`` packages:
+
+.. code-block:: bash
+
+    $ conan remove "*" -f
+    $ conan create pkga pkga/0.1@user/testing
+    $ conan create pkga pkga/0.1@user/testing -s build_type=Debug
+
+
+Now, we could (don't do it) create 2 different lockfiles, one for each configuration:
+
+
+.. code-block:: bash
+
+    # DO NOT type these commands, we'll do it better below
+    $ cd pkgb
+    $ conan lock create conanfile.py --user=user --channel=testing --lockfile-out=locks/pkgb_release.lock
+    $ conan lock create conanfile.py --user=user --channel=testing --lockfile-out=locks/pkgb_debug.lock -s build_type=Debug
+        
+
+.. important::
+
+    The dependency graph is different for each different configuration/profile. Not only the package-ids, but also because of
+    conditional requirements, the dependencies can be different. Then, it is necessary to create a lockfile for every different 
+    configuration/profile. 
+
+
+But what if between both commands, the new ``pkga/0.2@user/testing`` version was created? Although this is unlikely in this
+example, because everything is local, it is a possibility if ``pkga`` was in a server, and someone (or CI), upload the new
```

Comment:
Let's encourage the CI usage

```suggestion
example, because everything is local. However, it could happen that ``pkga`` was in a server and the CI, uploads the new
```